wp-includes/default-filters.php

```diff
@@ -17,7 +17,12 @@
 add_action( 'wp_ajax_nopriv_wp_service_worker', 'wp_ajax_wp_service_worker' );
 add_action( 'parse_query', 'wp_unauthenticate_error_template_requests' );
 
-add_action( 'wp_head', 'wp_add_error_template_no_robots' );
-add_action( 'error_head', 'wp_add_error_template_no_robots' );
+if ( version_compare( strtok( get_bloginfo( 'version' ), '-' ), '5.7', '>=' ) ) {
```

Comment:
Alternatively we could check for whether the `wp_robots` function exists to avoid the need for specific version lookup - but I guess the downside of that would be that someone else theoretically could declare a function of that name before. Just a thought, nothing blocking.